Upstream split this one commit into two: c9da8b0 and d6e25dd because
of some confusion.  I've recombined it here.  See:

https://bugzilla.gnome.org/show_bug.cgi?id=763567
https://bugs.gentoo.org/show_bug.cgi?id=577312

diff --git a/libs/gst/check/libcheck/strsignal.c b/libs/gst/check/libcheck/strsignal.c
index b79409b..57e71cd 100644
--- a/libs/gst/check/libcheck/strsignal.c
+++ b/libs/gst/check/libcheck/strsignal.c
@@ -1,6 +1,6 @@
 #include "libcompat.h"
 
-const char *
+char *
 strsignal (int sig)
 {
   static char signame[40];

diff --git a/libs/gst/check/libcheck/libcompat.h b/libs/gst/check/libcheck/libcompat.h
index 32f944c..f09289b 100644
--- a/libs/gst/check/libcheck/libcompat.h
+++ b/libs/gst/check/libcheck/libcompat.h
@@ -101,7 +101,7 @@ CK_DLL_EXP char *strdup (const char *str);
 #endif /* !HAVE_DECL_STRDUP && HAVE__STRDUP */
 
 #if !HAVE_DECL_STRSIGNAL
-CK_DLL_EXP const char *strsignal (int sig);
+CK_DLL_EXP char *strsignal (int sig);
 #endif /* !HAVE_DECL_STRSIGNAL */
 
 /*
